Mathlib/Topology/UniformSpace/UniformConvergenceTopology.lean

```diff
@@ -0,0 +1,942 @@
+uniform convergence. We denote it `α →ᵤ β`. -/
+def UniformFun (α β : Type _) :=
+  α → β
+#align uniform_fun UniformFun
+
+/-- The type of functions from `α` to `β` equipped with the uniform structure and topology of
+uniform convergence on some family `𝔖` of subsets of `α`. We denote it `α →ᵤ[𝔖] β`. -/
+@[nolint unusedArguments]
+def UniformOnFun (α β : Type _) (_ : Set (Set α)) :=
+  α → β
+#align uniform_on_fun UniformOnFun
+
+-- mathport name: «expr →ᵤ »
+scoped[UniformConvergence] notation:25 α " →ᵤ " β:0 => UniformFun α β
+
+-- mathport name: «expr →ᵤ[ ] »
+scoped[UniformConvergence] notation:25 α " →ᵤ[" 𝔖 "] " β:0 => UniformOnFun α β 𝔖
+
+-- mathport name: «exprλᵘ , »
+-- scoped[UniformConvergence] notation3"λᵘ "(...)", "r:(scoped p => UniformFun.ofFun p) => r
+
+-- mathport name: «exprλᵘ[ ] , »
+-- scoped[UniformConvergence] notation3"λᵘ["𝔖"] "(...)", "r:(scoped p => UniformFun.ofFun p) => r
+
+open UniformConvergence
+
+instance {α β} [Nonempty β] : Nonempty (α →ᵤ β) :=
+  Pi.Nonempty
+
+instance {α β 𝔖} [Nonempty β] : Nonempty (α →ᵤ[𝔖] β) :=
+  Pi.Nonempty
+
+/-- Reinterpret `f : α → β` as an element of `α →ᵤ β`. -/
+def UniformFun.ofFun {α β} : (α → β) ≃ (α →ᵤ β) :=
+  ⟨fun x => x, fun x => x, fun _ => rfl, fun _ => rfl⟩
+#align uniform_fun.of_fun UniformFun.ofFun
+
+/-- Reinterpret `f : α → β` as an element of `α →ᵤ[𝔖] β`. -/
+def UniformOnFun.ofFun {α β} (𝔖) : (α → β) ≃ (α →ᵤ[𝔖] β) :=
+  ⟨fun x => x, fun x => x, fun _ => rfl, fun _ => rfl⟩
+#align uniform_on_fun.of_fun UniformOnFun.ofFun
+
+/-- Reinterpret `f : α →ᵤ β` as an element of `α → β`. -/
+def UniformFun.toFun {α β} : (α →ᵤ β) ≃ (α → β) :=
+  UniformFun.ofFun.symm
+#align uniform_fun.to_fun UniformFun.toFun
+
+/-- Reinterpret `f : α →ᵤ[𝔖] β` as an element of `α → β`. -/
+def UniformOnFun.toFun {α β} (𝔖) : (α →ᵤ[𝔖] β) ≃ (α → β) :=
+  (UniformOnFun.ofFun 𝔖).symm
+#align uniform_on_fun.to_fun UniformOnFun.toFun
+
+-- Note: we don't declare a `CoeFun` instance because Lean wouldn't insert it when writing
+-- `f x` (because of definitional equality with `α → β`).
+end TypeAlias
+
+open UniformConvergence
+
+namespace UniformFun
+
+variable (α β : Type _) {γ ι : Type _}
+
+variable {s s' : Set α} {x : α} {p : Filter ι} {g : ι → α}
+
+/-- Basis sets for the uniformity of uniform convergence: `gen α β V` is the set of pairs `(f, g)`
+of functions `α →ᵤ β` such that `∀ x, (f x, g x) ∈ V`. -/
+protected def gen (V : Set (β × β)) : Set ((α →ᵤ β) × (α →ᵤ β)) :=
+  { uv : (α →ᵤ β) × (α →ᵤ β) | ∀ x, (uv.1 x, uv.2 x) ∈ V }
+#align uniform_fun.gen UniformFun.gen
+
+/-- If `𝓕` is a filter on `β × β`, then the set of all `uniform_convergence.gen α β V` for
+`V ∈ 𝓕` is a filter basis on `(α →ᵤ β) × (α →ᵤ β)`. This will only be applied to `𝓕 = 𝓤 β` when
+`β` is equipped with a `UniformSpace` structure, but it is useful to define it for any filter in
+order to be able to state that it has a lower adjoint (see `uniform_convergence.gc`). -/
```

Comment:
What are `uniform_convergence.gen` and `uniform_convergence.gc`? They don't exist in mathlib3 according to the docs -- was there a name change at some point...? Perhaps @ADedecker knows?

Reply:
I would guess that it is supposed to be `UniformFun.gen` and `UniformFun.gc`.

Reply:
Indeed I forgot to change the docstring